draft-bbs-signatures.md

```diff
@@ -534,45 +535,47 @@ Procedure:
 
 6. generators =  (H_s || H_d || H_1 || ... || H_L)
 
-7. domain = OS2IP(HASH(PK || L || generators || Ciphersuite_ID || header)) mod q
+7. revealed_msgs = (msg_i1 || msg_i2 || ... || msg_iR)
+
+8. domain = OS2IP(HASH(PK || L || generators || Ciphersuite_ID || header)) mod q
 
-8. for element in (r1, r2, e~, r2~, r3~, s~, m~_j1, ..., m~_jU):
+9. for element in (r1, r2, e~, r2~, r3~, s~, m~_j1, ..., m~_jU):
 
-9.      element = HASH(PRF(8*ceil(log2(q)))) mod q
+10.      element = HASH(PRF(8*ceil(log2(q)))) mod q
 
-10.      if element = 0, go back to step 7
+11.      if element = 0, go back to step 7
 
-11. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
+12. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
 
-12. r3 = r1 ^ -1 mod q
+13. r3 = r1 ^ -1 mod q
 
-13. A' = A * r1
+14. A' = A * r1
 
-14. Abar = A' * (-e) + B * r1
+15. Abar = A' * (-e) + B * r1
 
-15. D = B * r1 + H_s * r2
+16. D = B * r1 + H_s * r2
 
-16. s' = s + r2 * r3
+17. s' = s + r2 * r3
 
-17. C1 = A' * e~ + H_s * r2~
+18. C1 = A' * e~ + H_s * r2~
 
-18. C2 = D * (-r3~) + H_s * s~ + H_j1 * m~_j1 + ... + H_jU * m~_jU
+19. C2 = D * (-r3~) + H_s * s~ + H_j1 * m~_j1 + ... + H_jU * m~_jU
 
-19. c = HASH(PK || Abar || A' || D || C1 || C2 || ph)
+20. c = HASH(Abar || A' || D || C1 || C2 || R || revealed_msgs || domain || ph)
```

Comment:
We should add the messages too and not just the indices. We need them both. If that’s already the case it’s not clear to me how that is so given revealed_msgs appears to just include indices

Reply:
Thank you! Currently the PR only includes the revealed messages (note that `revealed_msgs =  (msg_i1 || msg_i2 || ... || msg_iR)` where `msg_i1, ..., msg_iR` the revealed messages in the current notation). Including the revealed indices also makes sense to me but there are proposals to remove the indices notion altogether (see for example issue #117 for some context). For that reason, I made the PR a draft for now, until a proper notation is decided.